### DISCUSSION TITLE
Feature Request / Documentation Help - Define Location of Static Pages as Something Other than the 'pages' folder

### DISCUSSION THREAD

**Discussion by princefishthrower** [gatsbyjs/gatsby#35727] [Ideas / Feature Requests] [answered]
As the title states, I'm unsure if it is possible to customize the folder Gatsby looks into to generate static pages. Apparently, this seems to be fixed without the ability to change from the `pages/` folder. Ideally, this could be changed with something like an environment variable or similar. On the existing docs (https://www.gatsbyjs.com/docs/creating-and-modifying-pages/) nothing like this is mentioned.

If it is not, it is a feature I'd love to have. My use case: deploying to Netlify and we have an `app` subdomain, where we would like all our static pages to be different than the static pages of our normal, non subomain deploy. This could be easily resolved simply by setting the default folder Gatsby look into to something like `app-pages/` instead of just `pages/`. 

Ultimately, I wouldn't care if either:

1. (Ideally) We can define what folder to create static pages from (hopefully not just `pages/`), or
2. I'm able to deactivate generation from `pages/` completely (not optimal, but then I can completely customize what I want over the `createPages()` API - cumbersome but should work - actually I guess this would already work but I would have to move my pages to some other folder)

**Comment by fbuireu** [comment, marked as answer]
Can be done by using[ `gatsby-plugin-page-creator`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-page-creator). You can easily customize the origin of the `/pages` folder by:

```
    {
      resolve: `gatsby-plugin-page-creator`,
      options: {
        path: `${__dirname}/src/app-pages`,
      },
    },
```

You can even have several instances of the plugin to split the page generation or ignore some files by using the `ignore` option:

```
    {
      resolve: `gatsby-plugin-page-creator`,
      options: {
        path: `${__dirname}/src/pages`,
        ignore: [`foo-bar.js`],
      },
    },
```

**Reply by princefishthrower** [reply]
Perfect, my final solution looked like this:

```javascript
const pageCreatorPluginEntries = [{
  resolve: `gatsby-plugin-page-creator`,
  options: {
    path: `${__dirname}/src/${process.env.STATIC_PAGES_DIRECTORY}`,
  },
}]

if (process.env.MY_CUSTOM_ENV === "MY_CUSTOM_ENV") {
  // need to ignore all pages in default gatsby page folder src/pages - add this additional override entry
  pageCreatorPluginEntries.push({
    resolve: `gatsby-plugin-page-creator`,
    options: {
      path: `${__dirname}/src/pages`,
      ignore: [`*`]
    },
  })
} 

module.exports = {
  plugins: [
    ...pageCreatorPluginEntries,
// more plugins and such here
  ]
}
```

Deployed to a Netlify subdomain and it's working epically!
